### DISCUSSION TITLE
Naming apparently is biased?

### DISCUSSION THREAD

**Discussion by dudemillion** [community/community#48096] [Repositories] [answered]
### Select Topic Area

Bug

### Body

So basically, I have a game made in Unity and are exporting it to GitHub. Now, the game was too big to put on Github.com directly, so I had to use GitHub Desktop to force the files onto GitHub, right? But there's a problem. One of my folders named Build (The folder Unity makes that contains the actual game) is not detected when put into the repo.  But when I name the folder anything BUT "Build" It recognizes it. Such as random blurbs like "fdajkg", "gajgdj", etc. I tried making the folder a random name so it would recognize it and then when it pushed, I would use GitHub ".dev" to change the name of it, but when I try to push to main after changing the name, it will always say "Unable to commit" Anyone know why this is happening? Because I sure as heck don't.

**Comment by spenserblack** [comment, marked as answer]
## Short Answer

Are you using the default Unity gitignore? [It ignores `Build`](https://github.com/github/gitignore/blob/4488915eec0b3a45b5c63ead28f286819c0917de/Unity.gitignore#L8), so git will *not* add the contents of this folder (except with `git add --force`).

## Long Answer

You probably shouldn't be pushing up built assets to GitHub, especially when these assets can be large. git isn't just a tool to back up your latest changes. It's a tool to *track change history.* Git can be really smart with source code -- if you add a line to a script, git will just track that one new line. So if you make a small change to a 10KB text file, your repository doesn't grow to 20 KB. But git *can't do this for binary files.* `.png`, `.exe`, `.zip` are some examples of binary files. If, for example, you have a 1GB `.exe`, you make a change and commit a new version, your repository *will* grow to 2GB. You should probably be uploading your built assets [to a GitHub Release](https://docs.github.com/en/repositories/releasing-projects-on-github/managing-releases-in-a-repository#creating-a-release). If you *must* store large files in your repository, you can look into [`git-lfs` (Large File Storage)](https://git-lfs.com/).
